asistente_ladm_col/lib/geometry.py

```diff
@@ -1112,3 +1113,22 @@ def get_dangle_ids(self, boundary_layer):
     def create_spatial_index(layer):
         if layer.hasSpatialIndex() != QgsFeatureSource.SpatialIndexPresent:
             processing.run("native:createspatialindex", {'INPUT': layer})
+
+    @staticmethod
+    def any_nan_z_coordinate(layer):
+        """
+        Check if layer has any z-coordinate with 'nan' value
+
+        :param layer: QgsVectorLayer
+        :return: True if any z-coordinate is 'nan', False otherwise
+        """
+        if layer.isSpatial():
+            # When the layer is not of type point or is multigeometry
+            # we extract the vertices to use the function z($geometry)
+            if layer.geometryType() != QgsWkbTypes.PointGeometry or QgsWkbTypes.isMultiType(layer.wkbType()):
+                layer = processing.run("native:extractvertices", {'INPUT': layer, 'OUTPUT': 'TEMPORARY_OUTPUT'})['OUTPUT']
+
+            request = QgsFeatureRequest(QgsExpression("z($geometry) = 'nan'")).setNoAttributes()
+            return bool([feature for feature in layer.getFeatures(request)])
```

Comment:
I'm here suggesting another way of checking the result. I'd like to avoid getting a Python list of thousands of `False` values (when using `getValues()`). I prefer to get only a list of features that have 'nan' values instead.

Could you check what works faster for a big Plot layer? Depending on that we'll make a decision.

Reply:
The performance test was performed with the two proposed algorithms (commit 7a467fe and ef70e64), the result obtained is as follows:

| **featureCount (plots)** | **Time in sec (7a467fe)** | **Time in sec (ef70e64)** |
| ------------------------ | ------------------------- | ------------------------- |
| 1493                     | 2.107273731               | 2.107357962               |
| 47776                    | 18.023672805              | 17.762942349              |
| 191104                   | 68.112640674              | 67.983314775              |
| 382208                   | 134.554609634             | 132.187930479             |

Reply:
Good, since differences are not dramatic (~1-2s in total), could we do a last test?

For the same cases, check the size in memory of the responses. That is, a set of few point features (vertices) with their attributes or a list of (hundred) thousands of `False` values.

Reply:
The ram memory consumption test was performed and the result is as follows:

| **featureCount (plots)** | **Size in bytes variable (7a467fe)** | **Maximum memory (MiB) usage function z_coordinate_is_nan (7a467fe)** | **Size in bytes variable (ef70e64)** | **Maximum memory (MiB) usage function z_coordinate_is_nan  (ef70e64)** |
| ------------------------ | ------------------------------------ | ------------------------------------------------- | ------------------------------------ | ------------------------------------------------- |
| 1493                     | 88                                   | 535.34765625                                      | 246080                               | 506.14453125                                      |
| 47776                    | 1240                                 | 1231.9453125                                      | 7872824                              | 1259.98828125                                     |
| 191104                   | 4264                                 | 3531.4140625                                      | 31491128                             | 3695.7265625                                      |
| 382208                   | 9016                                 | 6581.01171875                                     | 62982200                             | 6944.19140625                                     |

This exercise was very interesting, and it would be good to think in the future about the possibility of incorporating stress tests to measure the performance of certain algorithms.